js/tools/eraser.js

```diff
@@ -0,0 +1,28 @@
+/* global C2D */
+window.tools.eraser = {
+  name: 'eraser',
+  icon: '/images/eraser.png',
+  state: {
+    selected: false,
+    mousePressed: false
+  },
+  events: {
+    mousedown: function () {
+      const state = window.tools.eraser.state
+      state.mousePressed = true
+    },
+    mouseup: function () {
+      const state = window.tools.eraser.state
+      state.mousePressed = false
+    },
+    mousemove: function (e) {
+      const state = window.tools.eraser.state
+      if (state.selected && state.mousePressed) {
+        const mouse = C2D.eventToMouse(e)
+        const ctx = C2D.ctx
+        const width = window.options.strokeWidth.state.width
+        ctx.clearRect(mouse.x, mouse.y, width, width)
```

Comment:
this technically works, but i wonder if it would *feel* better to have the cursor in the center of the square (rather than the top left). by default the x,y coordinates u pass into `clearRect` will be the top left corner, but we can "offset" the top left corner by half the width `mouse.x - width / 2` so that the rectangle appears to surround the cursor's position.

Reply:
I just uploaded it into settings.json and also moved the cursor to the center, let me know if there's anything else I should checkout!